src/test/java/com/google/flourbot/api/DriveCloudSheetTest.java

```diff
@@ -0,0 +1,327 @@
+package com.google.flourbot.api;
+
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.Sheets.Spreadsheets;
+import com.google.api.services.sheets.v4.Sheets.Spreadsheets.Values;
+import com.google.api.services.sheets.v4.Sheets.Spreadsheets.Values.Get;
+import com.google.api.services.sheets.v4.model.ValueRange;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import static org.mockito.BDDMockito.when;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+@RunWith(JUnit4.class)
+public final class DriveCloudSheetTest {
+
+  // Create a sheet where in every cell A1:E5, the cell contents is the location
+  // Ex: The top left cell at A1 will contain the string "A1"
+  // Remember to give the bot permission to edit the sheet
+  private static final String SPREADSHEET_ID = "123";
+  private static final String SHEET_NAME = "Sheet1";
+
+  private static final String RANGE_SINGLE = "B1:B1";
+  private static final String RANGE_SINGLE_EMPTY = "F1:F1";
+  private static final String RANGE_BLOCK = "A1:C3";
+  private static final String RANGE_BLOCK_EMPTY = "G2:H4";
+  private static final String RANGE_ROW = "2:2";
+  private static final String RANGE_COLUMN = "C:C";
+
+  private static final int ROW = 3;
+  private static final String COLUMN = "D";
+
+  private static final int NUM_SMALL = 5;
+  private static final int NUM_BIG = 88;
+  
+  private Sheets sheetsServiceMock;
+  private Spreadsheets spreadsheetsMock;
+  private Values valuesMock;
+  private Get getMock;
+  private ValueRange valueRangeMock;
+  private CloudSheet cloudSheet;
+
+  @Before
+  public void setUp() {
+    sheetsServiceMock = Mockito.mock(Sheets.class);
+    spreadsheetsMock = Mockito.mock(Spreadsheets.class);
+    valuesMock = Mockito.mock(Values.class);
+    getMock = Mockito.mock(Get.class);
+    valueRangeMock  = Mockito.mock(ValueRange.class);
+    cloudSheet = new DriveCloudSheet(SPREADSHEET_ID, sheetsServiceMock);
+  }
+
+  @Test
+  public void testReadRangeSingle() {
+    String[] array = {"B1"};
+    List<List<Object>> values = Arrays.asList(
+      Arrays.asList(array)
+    );
+
+    when(sheetsServiceMock.spreadsheets()).thenReturn(spreadsheetsMock);
+    when(spreadsheetsMock.values()).thenReturn(valuesMock);
+    try {
```

Comment:
don't understand why the try is needed, but if it is needed please add Assert.fail in the catch block.